src/stores/models/CourseModel.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+
+import { observable } from 'mobx';
+import { ID } from '../../utils/id';
+
+export class CourseModel {
+  @observable name = '';
+  @observable id = '';
+  @observable dept = '';
+  @observable num = '';
+  @observable credits = 0;
+  @observable prereqs = [];
+
+  constructor(
+    name = 'A New Course',
+    dept = 'DEPT',
+    num  = '000',
+    credits = 3,
+    prereqs = [],
+  ) {
+    this.name = name;
+    this.dept = dept;
+    this.num = num;
+    this.credits = credits;
+    this.prereqs = prereqs;
+    this.id = ID();
```

Comment:
We have an ID and we calculate a course location every time we move a course - should we normalize our data? A course table that references a term PK and its index within that term, a Term table that references a Year PK (and its index), a Year table that references a Plan PK and its index, etc?

Reply:
If these were going to be huge tables, I'd agree with you, but reasonable use cases are going to be limited to just a few years, just two or three terms each, and up to maybe eight courses per term. We can totally consider it, but I'm reluctant to duplicate the information inherently in the state of the store because it easily becomes error prone when changes fail to update both representations.